docker/docker_linux.go

```diff
@@ -52,6 +52,58 @@ func GetDockerStat() ([]CgroupDockerStat, error) {
 	return ret, nil
 }
 
+// Generates a mapping of PIDs to container metadata.
+func GetContainerStats() (map[string]ContainerStat, error) {
+	containerMap := make(map[string]ContainerStat)
+	path := common.HostSys("fs/cgroup/cpuacct")
+	if common.PathExists(path) {
+		contents, err := common.ReadDirectory(path)
+		if err != nil {
+			return nil, err
+		}
+
+		// If docker containers exist, collect their stats.
+		if common.StringsContains(contents, "docker") {
+			dockerStats, err := GetDockerStat()
+			if err != nil {
+				return nil, err
+			}
+
+			for _, dockerStat := range dockerStats {
+				if !dockerStat.Running {
+					continue
+				}
+
+				dockerPids, err := CgroupPIDDocker(dockerStat.ContainerID)
+				if err != nil {
+					continue
+				}
+
+				bytes, err = json.Marshal(dockerStat)
+				if err != nil {
+					continue
+				}
+
+				var jsonMap map[string]string
+				err = json.Unmarshal(bytes, &jsonMap)
+				if err != nil {
+					continue
+				}
+				constainerStat := ContainerStat{
+					Type:     "Docker",
+					Metadata: jsonMap,
+				}
+
+				for _, pid := range dockerPids {
+					containerMap[pid] = constainerStat
```

Comment:
ideally there should be only 1 pid / container (https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#/run-only-one-process-per-container), i doubt this is strictly followed but i think the count should stay pretty low.